edb/common/parsing.py

```diff
@@ -142,49 +146,47 @@ def __new__(mcls, name, bases, dct):
 
                 a.__doc__ = attr.__doc__
                 a.inline_index = inline_index
-                setattr(result, name, a)
-
-        return result
+                setattr(cls, name, a)
 
 
-class Nonterm(parsing.Nonterm, metaclass=NontermMeta):
-    pass
-
-
-class ListNontermMeta(NontermMeta):
-    def __new__(mcls, name, bases, dct, *, element, separator=None):
-        if name != 'ListNonterm':
-            if issubclass(separator, Token):
-                separator = separator._token
-            elif isinstance(separator, NontermMeta):
-                separator = separator.__name__
+class ListNonterm(Nonterm, is_internal=True):
+    def __init_subclass__(cls, element, separator=None, is_internal=False,
```

Comment:
I would keep the `*, ` before `element`, which requires the arguments after it to be keyword arguments.

(That said, it doesn't really matter here, since there's not any way to pass non-keyword arguments to class creation)